### PR TITLE
Tutorial: Building a search engine on Kubernetes using Elasticsearch

### PR DESCRIPTION
**Pull request by aravindputrevu**
I'm adding a tutorial in the `stateful-application` section of Kubernetes Tutorials. 
The tutorial explains how to build a search appliance on Kubernetes step by step with some screenshots for guidance.

### CODE REVIEW

**Comment by aravindputrevu** [comment]
I have signed the CLA, just checking by changing the primary email.

**Comment by aravindputrevu** [comment]
@bradtopol / @zparnold - I have the same email in the CLA and commit, don't know what's wrong!

**Comment by jihoon-seo** [comment]
Preview links:
- https://deploy-preview-28837--kubernetes-io-main-staging.netlify.app/docs/tutorials/
- https://deploy-preview-28837--kubernetes-io-main-staging.netlify.app/docs/tutorials/stateful-application/elasticsearch/

File: content/en/docs/tutorials/stateful-application/elasticsearch.md [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+
+<!-- overview -->
+
+{{< warning >}}
+This deployment is not suitable for production use cases, you need to benchmark or size your cluster according to your use case.
+{{< /warning >}}
+
+{{< note >}}
+The files provided in this tutorial are using GA Deployment APIs and are specific to kubernetes version 1.16 and 1.20. You can also try it on newer Kubernetes version but some APIs might throw deprecated warnings. 
+{{< /note >}}
```
Comment (sftim) [inline content/en/docs/tutorials/stateful-application/elasticsearch.md:15]: Hi. These look like prerequisites - I'd mention them in the prerequisites section.

File: content/en/docs/tutorials/stateful-application/elasticsearch.md [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+The example shown on this page works with `kubectl` 1.16 and above.
+
+1. Brief understanding of REST APIs.
+2. High-level understanding of Elasticsearch.
+
+<!-- lessoncontent --> 
+
+# Creating an Elasticsearch Cluster
+
+There are many ways to run software like Elasticsearch on Kubernetes. For example, you could use the docker images or a helm chart. I’m using the official Kubernetes operator here.
```
Comment (sftim) [inline content/en/docs/tutorials/stateful-application/elasticsearch.md:35]: There's no official Kubernetes operator - I'm concerned that this wording might suggest that the operator is endorsed by Kubernetes. What I think you're describing is the official Elasticsearch operator (for deploying onto Kubernetes) from Elastic.

File: content/en/docs/tutorials/stateful-application/elasticsearch.md [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+The files provided in this tutorial are using GA Deployment APIs and are specific to kubernetes version 1.16 and 1.20. You can also try it on newer Kubernetes version but some APIs might throw deprecated warnings. 
+{{< /note >}}
+
+## {{% heading "objectives" %}}
+
+1. You will learn about deploying a stateful application Elasticsearch and a visualizer like Kibana on Kubernetes.
+2. Then, you will build a simple React-based search app using Kubernetes as a backend.
+
+
+## {{% heading "prerequisites" %}}
```
Comment (sftim) [inline content/en/docs/tutorials/stateful-application/elasticsearch.md:23]: How many nodes do you need? The tutorial doesn't make this clear.

**Review by sftim — COMMENTED** [review]
Thanks for proposing this @aravindputrevu 

It looks to me that this is documenting a specific technology (Elasticsearch). You could propose this tutorial for the Elasticsearch documentation instead; I think it would be a better fit there.

/hold
This doesn't seem to fit with the [content guide](https://kubernetes.io/docs/contribute/style/content-guide/)

**Comment by aravindputrevu** [comment]
@sftim I was thinking that we could document about running stateful apps as elasticsearch is one. So please let me know if I need to remove anything to make it better.

**Comment by sftim** [comment]
https://github.com/kubernetes/website/issues/16618 is a discussion broadly on this topic (improving on the existing stateful app tutorial). Maybe make a suggestion there?

**Comment by divya-mohan0209** [comment]
@aravindputrevu : Any progress on the issue/PR?

**Comment by aravindputrevu** [comment]
@divya-mohan0209 I have asked the same in parent issue. I have received no update.

**Comment by reylejano** [comment]
Hi @aravindputrevu , I agree that this PR on using Elasticsearch does not comply with the [Kubernetes content guide](https://kubernetes.io/docs/contribute/style/content-guide/) and is better suited in the Elasticsearch documentation.

I'm closing this PR but feel free to reopen this PR if you feel otherwise and we can discuss further.
/close

**Pull request closed without merging**